k8s/namespaces.go

```diff
@@ -53,6 +53,17 @@ func (kc *KubeClient) CreateOrUpdateNamespace(namespaceName string) (*corev1.Nam
 	return kc.Clientset.CoreV1().Namespaces().Update(ctx, nsSpec, metav1.UpdateOptions{})
 }
 
+// GetNamespace returns a kubernetes namespace object if it exists.
+func (kc *KubeClient) GetNamespace(namespaceName string) *corev1.Namespace {
+	ctx := context.TODO()
+		namespace, err := kc.Clientset.CoreV1().Namespaces().Get(ctx, namespaceName, metav1.GetOptions{})
+		if err != nil && k8sErrors.IsNotFound(err){
+			return nil
+		}
+
+	return namespace
```

Comment:
This will return `namespace` if the error is something else than Not Found, which will probably be `nil`. I think we should handle other errors.